test/functional/sqlsrv/sqlsrv_fetch_large_stream.phpt

```diff
@@ -0,0 +1,86 @@
+--TEST--
+Streaming Field Test
+--DESCRIPTION--
+Verifies the streaming behavior and proper error handling with Always Encrypted
+--SKIPIF--
+<?php require('skipif_versions_old.inc'); ?>
+--FILE--
+<?php
+require_once('MsCommon.inc');
+
+$conn = AE\connect();
+
+$tableName = "test_max_fields";
+dropTable($conn, $tableName);
+AE\createTable($conn, $tableName, array(new AE\ColumnMeta("varchar(max)", "varchar_max_col")));
+
+$inValue = str_repeat("ÃÜðßZZýA©", 600);
+$insertSql = "INSERT INTO $tableName (varchar_max_col) VALUES (?)";
+$params = array($inValue);
+
+$stmt = sqlsrv_prepare($conn, $insertSql, $params);
+if ($stmt) {
+    sqlsrv_execute($stmt);
+} 
+
+$query = "SELECT * FROM $tableName";
+$stmt = sqlsrv_prepare($conn, $query);
+if ($stmt) {
+    sqlsrv_execute($stmt);
+} 
+
+if (!sqlsrv_fetch($stmt)) { 
+    fatalError("Failed to fetch row ");
+}
+
+$stream = sqlsrv_get_field($stmt, 0, SQLSRV_PHPTYPE_STREAM(SQLSRV_ENC_CHAR));
+
+$success = false;
+if ($stream === false) {
+    $error = sqlsrv_errors()[0];
+    if (AE\isColEncrypted() && $error['SQLSTATE'] === "IMSSP" && $error['code'] === -109 &&
+        $error['message'] === "Connection with Column Encryption enabled does not support fetching stream. Please fetch the data as a string.") {
+        $success = true;
+    } else {
+        var_dump($error);
+        fatalError("Failed to read field");
+    }
+} else {
+    $value = '';
+    if (!AE\isColEncrypted()) {
+        $num = 0;
+        while (!feof($stream)) {
+            $value .= fread($stream, 8192);
+        }
+        fclose($stream);
+        if (checkData($value, $inValue)) {  // compare the data to see if they match!
+            $success = true;
+        } else {
+            fatalError("Data corruption!\n");
+        }
+    } else {
+        fatalError("Error in getting stream!");
+    }
+}
```

Comment:
perhaps simplify this checking because $stream is expected to fail with AE?

Reply:
use PHP_STREAM_BUFFER_SIZE?

Reply:
@yitam But the test should work without AE too right?
@lilgreenbird  where did you get that constant from? somehow when I put it in the test it returns undefined constant

Reply:
I just thought you could check if AE is enabled first and expected it to fail with the correct message?
Otherwise, it should pass with the results matched?